### PR TITLE
Fallback comparators (WIP)

### PR DESCRIPTION
**Pull request by mamonu**

### CODE REVIEW

File: splink/case_statements.py [hunk 1]
```diff
@@ -17,14 +17,30 @@ def _check_jaro_registered(spark):
...
 
     warnings.warn(
-        "Custom string comparison functions such as jaro_winkler_sim are available in"
-        " Spark Or you did not pass 'spark' (the SparkSession) into 'Model' You can"
-        " import these functions using the scala-udf-similarity-0.0.7.jar provided with"
-        " Splink"
+        "\n Custom string comparison functions such as jaro_winkler_sim are available in Spark"
+        "\n Or you did not pass 'spark' (the SparkSession) into 'Model' \n You can"
+        " import these functions using the scala-udf-similarity-0.0.9.jar provided with"
+        " Splink.\n" + _get_spark_jars_string()
```
Comment (RobinL) [inline splink/case_statements.py:23]: Do we want something to the effect of:
> If you're struggling with the jar, you can register a less efficient python implementatino of the jaro_winkler function with ```from splink.fallback import jc_sim_py; spark.udf.register("squaredWithPython", squared)

Comment (RobinL) [reply splink/case_statements.py:23]: Also need to register the functino as udf using 
```
#     spark.udf.registerJavaFunction(
#         "jaro_winkler_sim",
#         "uk.gov.moj.dash.linkage.JaroWinklerSimilarity",
#         types.DoubleType(),
#     )
```

**Comment by RobinL** [comment]
This is great.  If you're happy with the changes I just pushed, and the tests pass, happy to merge

image

File: splink/case_statements.py (hunk 1 above)
Comment (RobinL) [reply splink/case_statements.py:23]: I've now added this see below

**Comment by RobinL** [comment]
Just to note the changes I made:
- change filenames to `jar_fallback` for clarity
- added jar so now will be installed with Splink
- added detail to error message

**Comment by mamonu** [comment]
looks good to me.

**Pull request merged**